content/blog/clickhouse-benchmarking.md

```diff
@@ -0,0 +1,171 @@
+
+<p align="center">
+  <img src="/images/blog/clickhouse-benchmarking/clickhouse-read-operations.png" alt="clickhouse-benchmarking" style="border-radius: 10px; width: 300; height: 500;">
+</p>
+
+### Key Insights from the Data
+
+#### 1. `CPU Usage (avg_cpu_usage_cores)`
+
+- `Node-1` in the two-node configuration showed CPU usage between `0.574 to 0.875 cores`, with higher loads during complex query executions.
+- `Node-2` showed lower CPU usage overall, ranging between `0.493 to 0.122 cores` due to load balancing.
+- `Single Node configuration` saw the CPU usage fluctuate from `0.529 to 0.883 cores`, as it handled all query loads without load balancing support.
+
+#### 2. `Selected Bytes per Second (avg_selected_bytes_per_second)`
+
+- In the two-node configuration, `Node-1` selected bytes per second ranged from `143,598,799.18 to 224,235,858.8 B/s`, whereas `Node-2` showed lower selected bytes per second, from `80,651,591.63 to 22,842.02 B/s`, indicating an uneven load distribution.
+- In the `Single Node configuration`, this metric ranged from `158,795,019 to 283,619,700.4 B/s`, indicating the node’s capability to handle high data selection rates.
+
+#### 3. `I/O Wait (avg_io_wait)`
+
+- `Node-1` experienced minimal I/O wait, from `0.0002 to 0.0077 seconds`, while `Node-2` showed no I/O wait when handling lower loads.
+- The `Single Node configuration` maintained `0.0002 seconds` for I/O wait, reflecting efficient I/O handling even under increased loads.
+
+#### 4. `CPU Wait (avg_cpu_wait)`
+
+- `Node-1` in the two-node setup had a CPU wait ranging from `1.045 to 1.9356 seconds` during heavier query processing, while `Node-2` had a much lower CPU wait from `1.0964 to 0.0039 seconds`.
+- In `Single Node mode`, CPU wait times increased dramatically, ranging from `1.8338 to 9.1992 seconds`, indicating contention for CPU resources under increased query load.
+
+#### 5. `Disk Reads (avg_read_from_disk)`
+
+- Disk reads for `Node-1` ranged from `13,768,499.2 to 21,592,132.27 bytes` in the two-node configuration, with `Node-2` reading `7,768,951.47 to 0 bytes` depending on load distribution.
+- In `Single Node configuration`, disk reads increased significantly, varying from `3,258,299.73 to 11,590,451.2 bytes` under solo operation, revealing increased reliance on disk when a single node handled queries.
+
+#### 6. `Filesystem Reads (avg_read_from_filesystem)`
+
+- `Filesystem reads on Node-1` ranged from `14,377,664.77 to 22,227,520.87 bytes` in the two-node setup, while `Node-2` saw reads between `8,374,316.53 and 589,784.78 bytes`.
+- In `Single Node configuration`, filesystem reads ranged from `32,649,785.97 to 30,984,717.98 bytes`, showing a notable increase compared to the two-node configuration.
+
+#### 7. `Memory Usage (avg_memory_tracked)`
+
+- `Memory tracked` for `Node-1` ranged from `727,494,761.07 to 956,479,931 bytes`, and `Node-2` from `819,671,565.67 to 725,970,944 bytes` in the two-node setup.
```

Comment:
It would be great to use measurements that are more user-friendly and easier to understand. Why not consider using MB or GB instead?